servers/rendering/renderer_rd/shaders/taa_resolve.glsl

```diff
@@ -0,0 +1,402 @@
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#[compute]
+
+#version 450
+
+#VERSION_DEFINES
+
+// Based on Spartan Engine's TAA implementation https://github.com/PanosK92/SpartanEngine/blob/master/Data/shaders/temporal_antialiasing.hlsl
+
+#define USE_SUBGROUPS
+
+#define GROUP_SIZE 8
+#define FLT_MIN 0.00000001
+#define FLT_MAX 32767.0
+#define RPC_9 0.11111111111
+#define RPC_16 0.0625
+
+#ifdef USE_SUBGROUPS
+layout(local_size_x = GROUP_SIZE, local_size_y = GROUP_SIZE, local_size_z = 1) in;
+#endif
+
+layout(rgba16f, set = 0, binding = 0) uniform restrict readonly image2D color_buffer;
+layout(set = 0, binding = 1) uniform sampler2D depth_buffer;
+layout(rg16f, set = 0, binding = 2) uniform restrict readonly image2D velocity_buffer;
+layout(rg16f, set = 0, binding = 3) uniform restrict readonly image2D last_velocity_buffer;
+layout(set = 0, binding = 4) uniform sampler2D history_buffer;
+layout(rgba16f, set = 0, binding = 5) uniform restrict writeonly image2D output_buffer;
+
+layout(push_constant, std430) uniform Params {
+	vec2 resolution;
+	float camera_z_near;
+	float camera_z_far;
+}
+params;
+
+const ivec2 kOffsets3x3[9] = {
+	ivec2(-1, -1),
+	ivec2(0, -1),
+	ivec2(1, -1),
+	ivec2(-1, 0),
+	ivec2(0, 0),
+	ivec2(1, 0),
+	ivec2(-1, 1),
+	ivec2(0, 1),
+	ivec2(1, 1),
+};
+
+/*------------------------------------------------------------------------------
+						THREAD GROUP SHARED MEMORY (LDS)
+------------------------------------------------------------------------------*/
+
+const int kBorderSize = 1;
+const int kGroupSize = GROUP_SIZE;
+const int kTileDimension = kGroupSize + kBorderSize * 2;
+const int kTileDimension2 = kTileDimension * kTileDimension;
+
+vec3 reinhard(vec3 hdr) {
+	return hdr / (hdr + 1.0);
+}
+vec3 reinhard_inverse(vec3 sdr) {
+	return sdr / (1.0 - sdr);
+}
+
+float get_linear_depth(ivec2 thread_id) {
+	float near = params.camera_z_near;
+	float far = params.camera_z_far;
+	float z = 1.0 - texelFetch(depth_buffer, thread_id, 0).r;
+	float z_n = 2.0f * z - 1.0f;
+	return 2.0f * far * near / (near + far - z_n * (near - far));
+}
```

Comment:
This looks like it only handles perspective projection, we try to handle both perspective and orthogonal when we can for example: https://github.com/godotengine/godot/blob/8a21f7231f0851aec4c06abee28a549b1f8083fd/servers/rendering/renderer_rd/shaders/ssao.glsl#L129-L135